PortalFetch/crawler_sel.py

```diff
@@ -0,0 +1,240 @@
+    """Open search page."""
+    # Find the Apps menu and click
+    menus = driver.find_elements_by_class_name("list-group-item")
+    #print("menus: ", menus)
+    if not menus:
+        print("No left menu found！")
+        return
+    appMenu = []
+    for menu in menus:
+        txt = menu.text
+        if ("apps".lower() == txt.lower()): #  No left menu found
+            appMenu = menu
+            break
+    if not appMenu:
+        print("Apps menu not found in left menu！")
+        return
+    time.sleep(2)  # Wait for the next page to come over
+    appMenu.click()  # Open the menu
+    # Find in App List（Look Up Classes）button
+    clss = driver.find_elements_by_class_name("myapps-item")
+    if not clss:
+        print("No app list found！")
+        return
+    classes = []
+    for cls in clss:
+        txt = cls.find_element_by_class_name("myapps-item-label").text
+        txt = str.strip(txt)
+        #print("text:  ",  txt)
+        if ("Look Up Classes".lower() == txt.lower()):
+            classes = cls
+            break
+    if not classes:
+        print("No Look Up Classes feature found in the app list!")
+        return
+    mainWindowName = driver.window_handles[0]  # Main window name
+    classes.click()
+    time.sleep(2)  # Make sure the new window is open
+    #
+    # Switch to the newly opened window, because this window is the course search page, you must switch the program pointer to this window to operate this window
+    windowNames = driver.window_handles  # Get all window names
+    for name in windowNames:
+        if mainWindowName == name:
+            continue  # Do not switch to the main window because the search page opens from the main window
+
+        driver.switch_to_window(name)
+        break  # Because there are only two windows: the main window and the open window, switching directly to the end is over
+    # Waiting for elements in the page to appear, indicating that the page has finished loading
+    waitUtilPageLoaded(driver)
+
+
+# Go to the advanced options page and start filling in various search terms
+def fillAdvanceSearch(driver):
+    """Fill advance search."""
+    # Select all options in Subject list
+    subjectList = driver.find_element_by_id("subj_id")  # web element
+    subjectOptions = subjectList.find_elements_by_tag_name("option")  # list
+    subjectListSelect = Select(subjectList)
+    print("Start to select all the contents in the multi-selection drop-down box. There are many options. You may have to wait for more time. Don't worry ...")
+    for i in range(0, len(subjectOptions)):
+        subjectListSelect.select_by_index(i)
+
+    # Submit search
+    ipts = driver.find_elements_by_tag_name("input")  # list of Web Element
+    submit = ""  # webElement
+    for ipt in ipts:
+        if "submit" == ipt.get_attribute("type").lower() and \
+                "Section Search".lower() == ipt.get_attribute("value").lower():
+            submit = ipt
+            break
```

Comment:
Write a separate method to achieve this `for` loop could bring the Cognitive Complexity down to below 5